src/KnitPlugin.kt

```diff
@@ -103,6 +104,21 @@ open class KnitPluginExtension {
         ),
         files = files,
         rootDir = rootDir,
+        lineSeparator = evaluateLineSeparator(),
         check = check
     )
+
+    private fun evaluateLineSeparator(): String {
+        val ls = defaultLineSeparator
+        if (ls != null && ls != "\n" && ls != "\r\n" && ls != "\r") {
+            throw GradleException(
+                """Knit defaultLineSeparator must be one of:
+                |- Unix (\n)
+                |- Windows (\r\n)
+                |- Classic Mac OS (\r)
+            """.trimMargin()
+            )
+        }
+        return ls ?: System.lineSeparator()
```

Comment:
What puzzles me here is whether `System.lineSeparator()` should be the default here.

You've already done a great job with preserving separator for existing files, so the migration for existing users is gradual.

But having system-dependent separator as default behaviour may be not so convenient -- it means that every project using Knit and being developed on multiple platforms have to configure line separator.

WDYT about making UNIX's `\n` the default here?

Reply:
Thanks for looking at my PRs.

I think it is a good idea to set UNIX style as default.